crates/rune/src/compile/compile_visitor.rs

```diff
@@ -15,6 +15,13 @@ pub trait CompileVisitor {
 
     /// Visit something that is a module.
     fn visit_mod(&mut self, _source_id: SourceId, _span: Span) {}
+
+    /// Visit an item's doc comment attribute.
+    ///
+    /// This may be called several times for a single item. Each attribute should eventually be
+    /// combined for the full doc string.
+    fn visit_doc_comment(&mut self, _source_id: SourceId, _meta: MetaRef<'_>, _span: Span, _docstr: &str) {}
```

Comment:
Might be worth rewriting this in such a way where its called once with a collection of all doc comment attributes instead of being called for every attribute (span points to the whole set of comments and `docstr` becomes a `&Vec[&str]`).

Added benefit of immediately knowing how many comments there are for an item; this currently would have you only know at the end of compilation where you know that the function can't be called anymore.

Reply:
Not the worst thing in the world, but if you discover a better approach go for it!